tests/json.rs

```diff
@@ -1,22 +1,37 @@
 use super::*;
 
-fn case(justfile: &str, value: Value) {
+fn case<F: Fn(&Path) -> Value>(justfile: &str, value: F) {
```

Comment:
The JSON tests are pretty horrible. I changed the `case` function so that you pass in a struct which should serialize to the output of `--dump`.

You can take advantage of this by putting the relative path in the `source` field of the passed in `Module`, and then, in `case`, prepending the path to the temporary directory, which will make them match.

Reply:
Yeah, I agree. :-)